src/exit_codes.rs

```diff
@@ -21,6 +26,21 @@ impl ExitCode {
     fn is_error(self) -> bool {
         i32::from(self) != 0
     }
+
+    /// Exit the process with the appropriate code.
+    pub fn exit(self) -> ! {
+        #[cfg(unix)]
+        if self == ExitCode::KilledBySigint {
+            // Get rid of the SIGINT handler, if present, and raise SIGINT
+            unsafe {
+                if signal(Signal::SIGINT, SigHandler::SigDfl).is_ok() {
+                    let _ = raise(Signal::SIGINT);
+                }
+            }
```

Comment:
Hm, it's unfortunate that `ctrlc` doesn't provide a way to unregister the signal handler yet (see https://github.com/Detegr/rust-ctrlc/issues/28). If unregistering is that easy, do you think this could be upstreamed to `ctrl` somehow?

Reply:
I'll look in to this.  I like the suggested `ctrlc::Channel` API.